miden-tx/src/tests/kernel_tests/test_account.rs

```diff
@@ -620,11 +617,41 @@ fn test_authenticate_procedure() {
             root = prepare_word(&root)
         );
 
+        // Execution of this code would return an EventError(UnknownAccountProcedure) for procs
+        // that are not in the advice provider.
         let process = tx_context.execute_code(&code);
 
         match valid {
             true => assert!(process.is_ok(), "A valid procedure must successfully authenticate"),
-            false => assert!(process.is_err(), "An invalid procedure must fail to authenticate"),
+            false => assert!(process.is_err(), "An invalid procedure should fail to authenticate"),
         }
     }
 }
+
+#[test]
+fn test_procedure_out_of_bounds() {
+    let account_code = AccountCode::mock_wallet(TransactionKernel::assembler());
+
+    let tc_0: [Felt; 4] =
+        account_code.procedures()[0].mast_root().as_elements().try_into().unwrap();
+
+    let tx_context = TransactionContextBuilder::with_standard_account(ONE).build();
+
+    let code = format!(
+        "
+            use.kernel::account
+            use.kernel::prologue
+
+            begin
+                # authenticate procedure
+                push.{root}
+                exec.account::authenticate_procedure
+            end
+            ",
+        root = prepare_word(&tc_0)
+    );
```

Comment:
Why do we need `use.kernel::prologue` here? Seems like it is never used.

But also, it is not clear to me why this test works. We are trying to authenticate the first procedure in the mock wallet, right? We are able to find this procedure in the advice provider but somehow we get an incorrect index for it?

Reply:
This test works because we did not call `prologue::prepare_transaction` which means that the account procedures have not been read.

Hence `get_procedure_info` (in `authenticate_procedure`) will error out and return `ERR_PROC_INDEX_OUT_OF_BOUNDS` because the `num_account_procedures` is `0`.

I don't think that this test should be built in this way, it seems that we should: 
- Read the existing procedures using `prologue::prepare_transaction`
- Use an index that would be too far off
- Assert that the correct behaviour of erroring out of bounds was returned

It seems that we are testing an edge case / impossible case here

Reply:
Yeah, reviewing this, it does not make sense. I was initially trying to get to a point where I could get the error related to the proc root not being in the account code (instead of getting an event error for a procedure that is not in the advice provider).